shared/language-specs/spec.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * The specification used to provide search-based code intelligence for a
+ * particular language. This includes things like file extensions, comment
+ * patterns and delimiters, and logic for filtering out obviously wrong
+ * search results for definitions.
+ */
+export interface LanguageSpec {
+    /**
```

Comment:
I find the vertical displacement distracting here. Can we make one-liners of the ones that have only one line? E.g.,

```suggestion
/** Used to label markdown code blocks */
```

I'm pretty sure the doc comment parser is fine with this form.